Style Guide and Diataxis Review: Streams IV (#221)
iotaledger/streams: documentation/docs/overview.md

---

> description: "The Streams framework is intended to be a secure message verification and protection protocol for sending data over a given transport layer"

Reviewer: we don't need quotes here

- author
- subscriber
- value transactions
- explanation
---
# Overview

> The IOTA Streams framework is intended to be a secure message verification and protection protocol for sending data over a given transport layer. 

Reviewer: is intended makes it sound like it may not actually do it, wdyt?

> The Channels protocol is designed as a replacement for the MAM library. This was used for sending data using the Tangle as the primary transportation mechanism. The channels themselves can be structured in different ways with any arbitrary combination of publishers and subscribers (although each channel can only be hosted by a single author instance).

Reviewer: ```suggestion
The Channels protocol is a replacement for the MAM library. The MAM library was used for sending data using the Tangle as the primary transport mechanism. The channels can be structured differently with any arbitrary combination of publishers and subscribers (although a single author instance can only host one channel).
```

> # Channels Protocol

Reviewer: there should only be on h1 (level 1 header)
```suggestion
## Channels Protocol
```

Reviewer: please correct all subheader accordingly

The Channels protocol provides the high-level API tools necessary for authors and subscribers to be generated and interact with the Tangle. 

## Authors

> A channel author is responsible for both the generation and configuration of a new channel's structure (that is, single branch vs. multi-branch). An author of a channel will also be able to set the access restrictions to branches within a channel structure, as well as accepting and managing user subscription messages. 

Reviewer: ```suggestion
A channel author is responsible for generating and configuring a new channel's structure (that is, single branch vs. multi-branch). An author of a channel will also be able to set the access restrictions to branches within a channel structure, as well as accept and manage user subscription messages. 
```

## Subscribers 

> A channel subscriber is any user within a channel that is not the author. A subscriber can be generated independently without verification by an author. However, to write to a branch, or to process any private streams, they will be required to subscribe to the channel, and have the author accept and process that subscription. A subscriber may also use pre-shared keys as a method of interacting with a stream without going through the subscription process. 

Reviewer: ```suggestion
A channel subscriber is any user within a channel that is not the author. A subscriber can be generated independently without verification by an author. However, to write to a branch or to process any private streams, they will be required to subscribe to the channel and have the author accept and process that subscription. A subscriber may also use pre-shared keys to interact with a stream without going through the subscription process. 
```

## Branching

> Branches can be defined as any sequential grouping of messages that are linked to the announcement message. These branches will be generated with either a signed packet message or a keyload message for public or private streaming, respectively. A channel can also assume two different forms: 

Reviewer: ```suggestion
Branches are any sequential grouping of messages linked to the announcement message. You can generate branches with either a signed packet message or a keyload message for public or private streaming. A channel can also assume two different forms: 
```

- Single branch: a linear sequencing of messages (similar to a MAM stream) with each message linked to the previous one.
- Multi-branch: a sequencing of messages that do not rely on sequential linking of messages.

When generating a channel, the author will decide whether the channel will use single branching or multi-branching. This will inform the Streams instance of how it should perform sequencing. Subscribers will also be informed as they process the announcement message, so their instances know the appropriate sequencing order as well.   

## Keyloads

A keyload message is an access restriction and control message that allows the author to specify who should be able to decrypt any messages that are attached and are following it. There are two ways to determine access when generating a keyload message: 

> - Subscriber public keys.  

Reviewer: please add links to their respective headers

- Pre-shared keys. 

### Public Keys

> During the processing of subscription messages, public keys are masked and provided to the author for storage on their instance. That author can then specify which of these users will be able to access subsequent messages by including the public key in the keyload message.

Reviewer: ```suggestion
While processing subscription messages, public keys are masked and provided to the author for storage on their instance. That author can then specify which of these users will be able to access subsequent messages by including the public key in the keyload message.
```

### Pre-Shared Keys

> A pre-shared key is a predefined key shared between users using other means than the subscription process noted above. These keys can be used to provide access restrictions to a stream 
> without the need for a subscription process. 

Reviewer: ```suggestion
A pre-shared key is a predefined key shared between users using other means than the subscription process noted above. You can use these keys to provide access restrictions to a stream 
without the need for a subscription process. 
```

:::

## Sequencing

> Sequencing is the methodology built within streams that allows message identifiers to be sequentially generated regardless of the shape of the channel. Messages are identified by an indexation position within the Tangle. They are generated using several pieces of collected information that includes:

Reviewer: ```suggestion
Sequencing is the methodology built within streams that allows message identifiers to be sequentially generated regardless of the shape of the channel. An indexation position within the Tangle identifies messages. Message identifiers are generated using several pieces of collected information that include:
```

:::note

We recommend that a single branch be used with only a single publisher to avoid out of sync parties from 
issuing and retrieving messages from an incorrect sequencing combination (that is, Sub1 sends a message linked to Msg1 and Sub2 
sends a message linked to that same Msg1 before seeing and registering the message from Sub1, thus forking the stream). 
Multiple publishers can be used but it is important to ensure that they do not try to send messages in parallel to 
avoid this conflict. In the future, this may be an enforced limitation on single branch implementations to remove the 

> possibility altogether.

Reviewer: this seems a bit unclear to me.  Perhaps we could break it down a bit.
